Sources/Chat/ProtocolServices/History/HistoryService.swift

```diff
@@ -0,0 +1,37 @@
+import Foundation
+
+final class HistoryService {
+
+    private let historyClient: HistoryClient
+    private let seiralizer: Serializing
+
+    init(historyClient: HistoryClient, seiralizer: Serializing) {
+        self.historyClient = historyClient
+        self.seiralizer = seiralizer
+    }
+
+    func register() async throws {
+        try await historyClient.register(tags: ["2002"])
+    }
+
+    func fetchMessageHistory(thread: Thread) async throws -> [Message] {
+        let wrappers: [MessagePayload.Wrapper] = try await historyClient.getMessages(
+            topic: thread.topic,
+            count: 200, direction: .backward
+        )
+
+        return wrappers.map { wrapper in
+            let (messagePayload, messageClaims) = try! MessagePayload.decodeAndVerify(from: wrapper)
```

Comment:
should we force unwrap here?
would make sense to compactMap nil values or throw an error instead?